docs/framework/guides/deep-dive/localization.md

```diff
@@ -0,0 +1,240 @@
+
+- Adding support for localizing third-party plugins.
+- Allowing passing your own translations to fix missing or invalid localizations.
+- Generating deterministic builds.
+- Exposing easy-to-use APIs for providing translations and writing localizable content.
+- Supporting plural forms in each step of the localization system for more fluent translations.
+
+### Glossary of terms
+
+Before we start, let us explain the meaning of terms that are crucial for the translation process:
+
+- *A message* &ndash; A string or an object that should be translated.
+	The string version works as a shortcut for the `{ id: message, string: message }` object form.
+- *A message ID* &ndash; A property used to distinguish messages.
+	It is useful for short messages where a collision might occur, like `%0 images`.
+- *A message string* &ndash; The default (English) form of the message.
+	When the message is supposed to support plural versions, this is the default singular version.
+- *A message plural* &ndash; An optional plural (English) version of the message.
+	The presence of this property indicates that the message should support both singular and plural forms.
+- *A PO file (`.po`)* &ndash; A file containing translations for the language in the [PO format](https://www.gnu.org/software/gettext/manual/html_node/PO-Files.html).
+	All CKEditor 5 localizable packages contain such files in the `lang/translations/` directory.
+- *A POT file (`.pot`)* &ndash; A file containing source messages (English sentences) that will be translated.
+- *A translation asset* &ndash; A JavaScript file or part of the file with generated translations for one language.
+
+## Writing a localizable UI
+
+All *messages* that need localization should be passed to the special CKEditor 5's {@link module:utils/locale~Locale#t `t()` function}. This function can be retrieved from the editor's {@link module:utils/locale~Locale} instance: `const { t } = editor.locale;` or from any view method: `const t = this.t;`.
+
+As the first argument, the `t()` function accepts either a string literal or an object literal containing the `id`, `string` and `plural` (optional) properties. The string literal will serve as both the *message ID* and the *message string*.
+
+As the second argument, the translation function accepts a value or an array of values. These values will be used to fill the placeholders in more advanced translation scenarios. If the `plural` property is specified, the first value will be used as the quantity determining the plural form.
+
+When using the `t()` function, you can create your own *localizable messages* or reuse *messages* created in CKEditor 5 packages that your project depends on. In case of reusing *messages*, you will not need to worry about translating them as all work will be done by the CKEditor 5 team and [Transifex translators](https://www.transifex.com/ckeditor/ckeditor5/) anyway. Obviously, your help in translating will still be appreciated!
```

Comment:
```suggestion
When using the `t()` function, you can create your own *localizable messages* or reuse *messages* created in CKEditor 5 packages that your project depends on. In case of reusing *messages*, you will not need to worry about translating them as all work will be done by the CKEditor 5 team and [Transifex translators](https://www.transifex.com/ckeditor/ckeditor5/). Obviously, your help in translating will still be appreciated!
```